src/app/client/src/app/modules/dashboard/components/report/report.component.ts

```diff
@@ -0,0 +1,310 @@
+import { IImpressionEventInput } from '@sunbird/telemetry';
+import { INoResultMessage, ResourceService, ToasterService, NavigationHelperService } from '@sunbird/shared';
+import { ActivatedRoute, Router } from '@angular/router';
+import { UserService } from '@sunbird/core';
+import { Component, OnInit, ViewChildren, QueryList, ViewChild, AfterViewInit } from '@angular/core';
+import { ReportService } from '../../services';
+import * as _ from 'lodash-es';
+import { Observable, throwError, of, forkJoin, combineLatest } from 'rxjs';
+import { mergeMap, switchMap, map, retry, catchError, tap } from 'rxjs/operators';
+import { DataChartComponent } from '../data-chart/data-chart.component';
+import html2canvas from 'html2canvas';
+import * as jspdf from 'jspdf';
+import { UUID } from 'angular2-uuid';
+
+@Component({
+  selector: 'app-report',
+  templateUrl: './report.component.html',
+  styleUrls: ['./report.component.scss']
+})
+export class ReportComponent implements OnInit, AfterViewInit {
+
+  public report: any;
+  public report$;
+  public noResultMessage: INoResultMessage;
+  public noResult: boolean;
+  private downloadUrl: string;
+  public reportObj: any;
+  telemetryImpression: IImpressionEventInput;
+  @ViewChildren(DataChartComponent) chartsComponentList: QueryList<DataChartComponent>;
+  @ViewChild('reportElement') reportElement;
+  public hideElements: boolean;
+  public reportExportInProgress = false;
+
+  constructor(private reportService: ReportService, private userService: UserService, private activatedRoute: ActivatedRoute,
+    private resourceService: ResourceService, private toasterService: ToasterService,
+    private navigationhelperService: NavigationHelperService,
+    private router: Router) { }
+
+  ngOnInit() {
+    this.report$ = this.activatedRoute.params.pipe(
```

Comment:
does this page change its route internally?. If no then we can use activatedRouteSnapshot to get reportId. 
And start from this.reportService.isAuthenticated instead switchMap is not necessary.

Reply:
Done